vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_embedding_info.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/ml/pipeline/pipeline_embedding_info.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/time/time.h"
+
+namespace ads {
+namespace ml {
+namespace pipeline {
+
+EmbeddingPipelineInfo::EmbeddingPipelineInfo() = default;
+
+EmbeddingPipelineInfo::EmbeddingPipelineInfo(
+    EmbeddingPipelineInfo&& info) noexcept = default;
+
+EmbeddingPipelineInfo& EmbeddingPipelineInfo::operator=(
+    EmbeddingPipelineInfo&& info) noexcept = default;
+
+EmbeddingPipelineInfo::~EmbeddingPipelineInfo() = default;
+
+bool EmbeddingPipelineInfo::FromValue(const base::Value::Dict& root) {
+  if (absl::optional<int> value = root.FindInt("version")) {
+    version = value.value();
+  } else {
+    return false;
+  }
+
+  if (const auto* value = root.FindString("timestamp")) {
```

Comment:
clang-tidy will be updated soon to not use magic strings/numbers. can you please move to `constexpr` in an anonymous namespace. Please check other usage.